testutils/infra/container_manager/docker_compose_manager.py

```diff
@@ -346,12 +348,17 @@ def __init__(self, name, num_clients=0):
             )
 
     def setup(self, recreate=True, env=None):
+        self.smtp_mock_lock.acquire(timeout=20 * 60)
         cmd = "up -d"
         if not recreate:
             cmd += " --no-recreate"
         self._docker_compose_cmd(cmd, env=env)
         self._wait_for_containers(self.NUM_SERVICES_ENTERPRISE + 1)
 
+    def teardown(self):
+        super().teardown()
+        self.smtp_mock_lock.release()
```

Comment:
Nitpick: Construction/destruction semantics normally require a LIFO approach, in other words the `super.teardown()` should be called last. Unlikely to be important here, but it's good style nevertheless.

Reply:
I would normally do LIFO approach too, but in this particular case it can lead to error: the lock would be released before calling `docker-compose down`, and then the next test could potentially call `docker-compose up` while the port is still in use.

Reply:
You're right. And actually, your approach *is* LIFO, I just missed that in the construction, the `super` object is constructed last. 🙂